lib/ice_cube/builders/ical_builder.rb

```diff
@@ -37,11 +37,13 @@ def self.ical_utc_format(time)
 
     def self.ical_format(time, force_utc)
       time = time.dup.utc if force_utc
+
       if time.utc?
         ":#{IceCube::I18n.l(time, format: '%Y%m%dT%H%M%SZ')}" # utc time
+      elsif time.respond_to?(:time_zone)
+        ";TZID=#{time.time_zone.name}:#{IceCube::I18n.l(time, format: '%Y%m%dT%H%M%S')}" # time zone specified
       else
-        time_zone = time.respond_to?(:time_zone) ? time.time_zone.name : time.zone
-        ";TZID=#{time_zone}:#{IceCube::I18n.l(time, format: '%Y%m%dT%H%M%S')}" # local time specified
+        ":#{IceCube::I18n.l(time, format: '%Y%m%dT%H%M%S')}" # local time specified
```

Comment:
As shown in the spec, dates in local / floating time should be serialized in the form `YYYYMMDDTHHMMSS`. Previously they were serialized with `TZID=${time.zone}`.

While I don't think `TZID=${time.zone}` is invalid (if only because, as the spec states, there is no specified format for timezones), it isn't a *local* time and it also doesn't produce time zones which ActiveSupport expects.